### DISCUSSION TITLE
SftpPermissionDeniedException when deleting multiple files async

### DISCUSSION THREAD

**Discussion by PeterKaspar** [sshnet/SSH.NET#1428] [Q&A]
I have a piece of code that recursively iterates over a folder, and deletes all found files. I have written it async to make sure it runs as fast as possible, but I keep getting SftpPermissionDeniedException. Does this mean SftpClient does not support multiple async delete tasks, and I have to await each delete task before starting the next one? 

```c#
        await _pipeLine.ExecuteAsync(async ctoken => await client.ConnectAsync(ctoken), ct);

        await _pipeLine.ExecuteAsync(async ctoken =>
        {
            var deleteTasks = new List<Task>();
            await foreach(var file in RecursiveIterateFilesAsync(client, _rootFolder, ct).WithCancellation(ct))
            {
                        deleteTasks.Add(client.DeleteFileAsync(file.FullName, ct));
            }
            await Task.WhenAll(deleteTasks); //is this supported by SftpClient? or do I have to await each delete task before starting the next one?
        }, ct);
```

**Comment by Rob-Hague** [comment]
Have you tried awaiting each one in turn, to see if that's the problem?

**Comment by PeterKaspar** [comment]
Allright, the code does work, the problem was that the user I logged in with, actually didn't have delete permissions... But the SFTP connection does support multiple async calls at the same time. Nice to know!
